src/transport/emitter.ts

```diff
@@ -0,0 +1,64 @@
+import { CloudEvent } from "../event";
+import { emitBinary, emitStructured } from "./http";
+import { Protocol } from ".";
+
+export interface Options {
+  /**
+   * The endpoint that will receieve the event.
+   * @example http://cncf.example.com/receiver
+   */
+  url?: string;
+  /**
+   * The network protocol over which the event will be sent.
+   * @example HTTPStructured
+   * @example HTTPBinary
+   */
+  protocol?: Protocol;
+
+  [key: string]: string |  object | Protocol | undefined;
+}
+
+/**
+ * A class to send binary and structured CloudEvents to a remote endpoint.
+ * Currently, supported protocols are HTTPBinary and HTTPStructured.
+ *
+ * @see https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md
+ * @see https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md#13-content-modes
+ */
+export class Emitter {
```

Comment:
nit: a name like `CloudEventEmitter` could be more descriptive and help with search.